tools/nvqpp/nvq++.in

```diff
@@ -242,13 +243,31 @@ LIBRARY_MODE=true
 CXX=${LLVMBIN}clang++${llvm_suffix}
 LINKER_CXX=${CXX}
 CMAKE_FALLBACK_HOST_CXX=${CXX}
-COPY_INPUT_ARGS=$@
+TARGET_ARGS=$@
 
 if [[ $# -eq 0 ]]; then
     SHOW_HELP=true
 fi
 
+# First, find the target option. The name is needed since it is the prefix for
+# any target specific options.
+for (( i=1; i<= $#; i+=1 )); do
+	if [[ ${@:$i:1} == --target ]] || [[ ${@:$i:1} == -target ]]; then
+		TARGET_CONFIG="${@:$i+1:1}"
+		break
+	fi
+done
+
 while [ $# -ne 0 ]; do
+	# Filter out any target-specific options.
+	if [[ "$1" == --${TARGET_CONFIG}-* ]]; then
+		# Assume always has an additional value
```

Comment:
I think it might be worth copying this assumption (always exactly 1 parameter and it must start with `--${targetName}-`) into the `{ionq,iqm,oqc,quantinuum}.config` files because as we onboard new providers, that's probably the place they will look for reference about how to add new parameters, and they probably won't realize that they have those limitations unless we make it clear to them.

Reply:
Sure.